### DISCUSSION TITLE
1.parse date-and-time node(lyd_parse_op)(2023-06-02T20:13:25Z);2.  print this node(lyd_print_mem), the value change to 2023-06-03T04:13:25+08:00. Is there some method to refuse this change?

### DISCUSSION THREAD

**Discussion by jiayinhaoran** [CESNET/libyang#2062] [Q&A]
1.  parse date-and-time node(lyd_parse_op)(2023-06-02T20:13:25Z);
{"xxx:date-time":"2023-06-02T20:13:25Z"}
use these interfaces:
ly_in_new_memory
lyd_parse_op

2.  print this node(lyd_print_mem), the value change to 2023-06-03T04:13:25+08:00
use these interface:
lyd_print_mem
the result is: {"xxx:date-time":"2023-06-03T04:13:25+08:00"}

Why GMT time is automatically changed to local time?
I find that 'lyplg_type_print_date_and_time' will print data-and-time type node value, and it look like do a forced conversion.
I also read this link: https://github.com/CESNET/libyang/issues/2045
But, I think that:
![image](https://github.com/CESNET/libyang/assets/11779997/1754731b-12d1-42c7-b101-30783a95b95b)
from rfc3339, data-time's  time-offset   can be  "Z" or time-numoffset, why must print with time-numoffset?

Is there some method to refuse this change or how can I print date-and-time with "Z" time?
@michalvasko

**Comment by michalvasko** [comment]
YANG `date-and-time` has its own [specification](https://www.rfc-editor.org/rfc/rfc6991.html#page-11) and rules. To avoid using timezone-specific timestamps, you need to use the special timezone `-00:00`.
